### PR TITLE
Fix null ref in change signature.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/65298
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1645107

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@dibarbet this was introduced in https://github.com/dotnet/roslyn/pull/63546.  Sorry for the lack of NRT annotations to help catch this.

**Comment by CyrusNajmabadi** [comment]
@akhera99 ptal.

File: src/Features/Core/Portable/ChangeSignature/AbstractChangeSignatureService.cs [hunk 1]
```diff
@@ -1059,17 +1058,15 @@ protected static int GetParameterIndexFromInvocationArgument(SyntaxNode argument
...
-
-            if (parameter.ContainingSymbol is IMethodSymbol methodSymbol && methodSymbol.MethodKind is MethodKind.ReducedExtension)
-            {
-                // We're in the invocation of an extension method that is called via this.Method(params).
-                // The 'this' argument has an ordinal value of -1 but change signature is expecting all params
-                // to start at 0 (including the 'this' param).
-                parameterIndex += 1;
-            }
+            if (parameter is null)
+                return 0;
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/ChangeSignature/AbstractChangeSignatureService.cs:1062]: this is the crash fix.

File: src/Analyzers/CSharp/Analyzers/UseIndexOrRangeOperator/CSharpUseIndexOperatorDiagnosticAnalyzer.cs [hunk 2]
```diff
@@ -194,7 +195,10 @@ protected override void InitializeWorker(AnalysisContext context)
                 return;
             }
 
-            if (CSharpSemanticFacts.Instance.IsInExpressionTree(instance.SemanticModel, instance.Syntax, infoCache.ExpressionOfTType, cancellationToken))
+            var semanticModel = instance.SemanticModel;
+            Contract.ThrowIfNull(semanticModel);
```
Comment (CyrusNajmabadi) [inline src/Analyzers/CSharp/Analyzers/UseIndexOrRangeOperator/CSharpUseIndexOperatorDiagnosticAnalyzer.cs:199]: NRT was turned on for a bunch of files.  so there's ancillary fallout of htat in this PR.

**Review by akhera99 — APPROVED** [review]

File: src/Analyzers/CSharp/Analyzers/UseIndexOrRangeOperator/CSharpUseIndexOperatorDiagnosticAnalyzer.cs (hunk 2 above)
Comment (akhera99) [reply src/Analyzers/CSharp/Analyzers/UseIndexOrRangeOperator/CSharpUseIndexOperatorDiagnosticAnalyzer.cs:199]: Lgtm besides the rest of the null references that need fixing.

**Pull request merged**